packages/vue-component-library/src/lib-components/FilterSelections.vue

```diff
@@ -0,0 +1,273 @@
+    return []
+
+  const filterKey = getFilterKey(filter)
+  const isFiltered = filteredStates.value[filterKey]
+
+  const options = isFiltered
+    ? filter.options.filter((option: any) => selectedOptions.value[filterKey]?.includes(option.value))
+    : filter.options
+  return options
+}
+
+// Helper function for template binding
+function createToggleOptionHandler(filterName: string) {
+  return (option: FilterOption) => toggleOption(filterName, option)
+}
+
+// Toggle option selection
+async function toggleOption(filterName: string, option: FilterOption) {
+  const currentSelection = selectedOptions.value[filterName] || []
+  const optionIndex = currentSelection.indexOf(option.value)
+
+  if (optionIndex > -1) {
+    // Remove option
+    currentSelection.splice(optionIndex, 1)
+    emit('option-deselected', filterName, option)
+
+    if (currentSelection.length === 0) {
+      const filterIndex = getFilterIndex(filterName)
+
+      if (filterIndex >= 0 && toggleRefs.value[filterIndex]) {
+        if (filteredStates.value[filterName]) {
+          // Toggle is in filtered mode, close it and reset state
+          await nextTick()
+          toggleRefs.value[filterIndex].close()
+          filteredStates.value[filterName] = false
+        }
+        else {
+          // Toggle is not in filtered mode, just reset the filtered state but don't close
+          filteredStates.value[filterName] = false
+        }
+      }
+    }
+    else {
+      // Animate height if needed
+      const filterIndex = getFilterIndex(filterName)
+      if (filterIndex >= 0 && toggleRefs.value[filterIndex]?.isOpened && filteredStates.value[filterName])
+        await animateHeightChange(filterIndex)
+    }
+  }
+  else {
+    // Add option
+    currentSelection.push(option.value)
+    emit('option-selected', filterName, option)
+  }
+
+  // Update router query parameters based on current selections
+  const nextQuery: Record<string, any> = { ...route.query }
+
+  // For each filter, map selections to its facetField in query
+  props.filters.forEach((filter) => {
+    const key = getFilterKey(filter)
+    const facet = filter.facetField
+    if (!facet)
+      return
+
+    const selections = selectedOptions.value[key] || []
+    if (selections.length > 0)
+      nextQuery[facet] = selections
+    else
+      delete nextQuery[facet]
+  })
+
+  // Reset pagination param if present
+  if ('page' in nextQuery)
+    delete nextQuery.page
+
+  router.push({ query: nextQuery })
+}
+</script>
+
+<template>
+  <div :class="classes">
+    <div class="filter-list">
+      <EffectSlideToggle
+        v-for="filter in props.filters"
+        :key="filter.name"
+        ref="toggleRefs"
+        class="filter-item"
+        :prevent-close="hasSelectedOptions(getFilterKey(filter))"
+      >
+        <!-- SUMMARY -->
+        <template #summary>
+          <div
+            class="filter-summary"
+            :class="{ 'is-filtered': filteredStates[getFilterKey(filter)] }"
```

Comment:
You can move that logic into a function:

function summaryClasses(filterKey) {
  return {
    
    'is-filtered': filteredStates.value[filterKey],
  
  }
}


Then in your template:
```
<div :class="summaryClasses(getFilterKey(filter))">
  ...
</div>
```